### PR TITLE
Fix #26857: separate stdout from stderr in AzureCliCredential

### LINKED ISSUE #26857: AzureCliCredential fails due to Python deprecation message getting mixed with token output
- **Package Name**: azure-identity
- **Package Version**: 1.10.0
- **Operating System**: WSL
- **Python Version**: 3.6.0

**Describe the bug**
In https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py#L137, the option `"stderr": subprocess.STDOUT` makes `check_output` also include stderr in the returned string. I was trying to use this package in a container with a somewhat out-of-date python setup, which caused `az account get-access-token` to output some warnings to stderr:
RequestsDependencyWarning from python screenshot

Since `_run_command`'s return value is parsed as JSON, this caused an exception and resulted in a somewhat confusing error message:

```
azure.core.exceptions.ClientAuthenticationError: DefaultAzureCredential failed to retrieve a token from the included credentials.
...
```

**To Reproduce**
(Environment-specific, but reasoning for this issue described above)

**Expected behavior**
`AzureCliCredential` should work

### PR DESCRIPTION
**Pull request by micromaomao**
This PR fixes https://github.com/Azure/azure-sdk-for-python/issues/26857 in the way proposed in https://github.com/Azure/azure-sdk-for-python/issues/26857#issuecomment-1281134407. Essentially, AzureCliCredentials now treats stderr separately from stdout, and therefore if for some reason the call to `az` was successful but with some warnings, AzureCliCredentials will still work.

I have manually tested that the various error code paths still works like it does currently, by using a mocked `az` binary.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by pvaneck — COMMENTED** [review]
Thanks for the PR!

Looks like users will also hit a similar issue with the [async version](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py#L103) of `AzureCLICredential`.

Do you mind verifying with something like:

```python
import asyncio
from azure.identity.aio import AzureCliCredential

async def test():
    credential = AzureCliCredential()
    await credential.get_token("https://management.core.windows.net")

asyncio.run(test())
```
and applying similar changes to the async version if it fails?

**Comment by micromaomao** [comment]
Hi @pvaneck - this has been addressed, please review.

File: sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py:108]: Why we need this?

File: sdk/identity/azure-identity/CHANGELOG.md
Comment (pvaneck) [inline sdk/identity/azure-identity/CHANGELOG.md:11]:
```suggestion
* `AzureCliCredential` now works even when `az` prints warnings to stderr. ([#26857](https://github.com/Azure/azure-sdk-for-python/issues/26857))
```

**Review by pvaneck — APPROVED** [review]
Thanks for adding the fix for the async credential!

Initially, was thinking it may be useful to log the stderr in cases where the `az` returncode is `0` but stderr is not `None`, though I am not sure how beneficial/necessary that would be. So, I think this looks good. :)

I'll wait for @xiangyan99's approval before merging.

File: sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py
Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py:108]: I see you have

output = stdout_b.decode()
stderr = stderr_b.decode()

and seems like now we only use stderr.

Is there any reason to keep output = stdout_b.decode()?

Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py:108]: output is the actual return value on line 119 :)

Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py:108]: Why can't we return stderr? :)

They are same, no?

Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/aio/_credentials/azure_cli.py:108]: From what I understand, stdout contains the actual access token json output returned by the az cli, while stderr will naturally have the error messages and sometimes warnings as in the tagged issue's case.

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**
